src/config/setup-test-container.mjs

```diff
@@ -51,7 +51,10 @@ export function setupTestContainer() {
     // wrap it in a spy so we can record calls, but use the
     // real implementation
     parseYargs: awilix.asValue(spy(parseYargs)),
-    fs: awilix.asValue(stub(fs)),
+    // Stubbing node:fs globally breaks tests where we want fs to work normally.
+    // Let tests decide how fs should be stubbed out if they don't want the real
+    // implementation.
+    fs: awilix.asValue(fs),
```

Comment:
i actually think we should invert this - you can re-register a dependency to overwrite it, so tests that need the real fs implementation can clobber the stub like:
```
// in test
before(() => {
  container.register({ fs: awilix.asValue(fs) });
});
```

most tests should mock fs - they'll be faster and more resilient that way. shell history is an outlier because we don't control the code that does the writing - but in general, we should prefer to mock (and making mocking the default behavior makes it more likely that later devs will choose that).

Reply:
I'll give it a shot.